### PR TITLE
feat: Blockquote component

### LINKED ISSUE #5820: Create `Blockquote` Component
Create a `Blockquote` component using the new [figma design (direct link)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=1509-16204&mode=dev). 

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).
- Review the [component creation guide](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#creating-components) to follow our standards.

---

### Details

image

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)

- Create a new directory called `Blockquote` within the [`Common` directory](https://github.com/nodejs/nodejs.org/tree/main/components/Common) that contains all the new code.
- Use the new tailwind color and typography system to assign properties. For example, `Light` uses `Green/600` per the figma.
- Add a storybook story file `components/Common/Blockquote/index.stories.tsx` which exercises each of the button's states.
- Add the em-dash `—` for users before the `attribution` prop renders

Suggested props include:

- `children` text content, rendered within a `p` tag: `“In a world older and more complete than ours...` in the figma
- `attribution` (string) : `Hayden Bleasel, Stoic Philosopher` in the figma

There are 2 states to capture within styles and stories:

- Light
- Dark

### Open Questions, to Discuss

- how should the semantic markup be structured? this is a topic of much debate, it seems. suggest starting with [MDN's approach](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blockquote)
- how does a user-supplied markdown blockquote `> text` render in tailwind? that will certainly be something that comes up. I suggest we add a storybook story also rendering this (but not using the `Blockquote` component, as markdown-driven content won't have `attribution` in many cases. I don't think an MDX component map is appropriate right now either.

### PR DESCRIPTION
**Pull request by canerakdas**
## Description

This pull request addresses the issue(#5820)  by adding a new directory called Blockquote within the Common directory. This new directory contains all the code necessary for the Blockquote component.

### Changes

- Created a new directory: `Common/Blockquote`.
- Implemented the Blockquote component using the new Tailwind color, typography system, and Figma design.
- Added a Storybook story file: `components/Common/Blockquote/index.stories.tsx`, which exercises each of the Blockquote component's states.
	- Default
	- Without Attribution
	- Nested

## Validation

image
image

## Related Issues

fixes #5820 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/Common/Blockquote/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
+
+import styles from './index.module.css';
+
+type BlockquoteProps = ComponentProps<'blockquote'>;
+
+const Blockquote: FC<PropsWithChildren<BlockquoteProps>> = ({
+  children,
+  cite,
```
Comment (ovflowd) [inline components/Common/Blockquote/index.tsx]: What exactly is this cite prop? Is this part of the native Blockquote HTML element?

File: components/Common/Blockquote/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import type { Meta as MetaObj, StoryObj } from '@storybook/react';
+
+type Story = StoryObj<typeof Blockquote>;
+type Meta = MetaObj<typeof Blockquote>;
+
+export const Default: Story = {
+  args: {
+    children: (
+      <>
+        <p>
```
Comment (ovflowd) [inline components/Common/Blockquote/index.stories.tsx]: You don't need to pass paragraphs as children here, can you remove these <p> elements to simplify things?

MDX sometimes will make paragraph elements but not all the times

**Review by ovflowd — APPROVED** [review]
LGTM! Thank you for the awesome contribution!

I also left a few comments :)

File: components/Common/Blockquote/index.tsx (hunk 1 above)
Comment (canerakdas) [reply components/Common/Blockquote/index.tsx]: Yes, it's an attribute of the Blockquote HTML element. I couldn't figure out exactly what it does.

**Comment by mikeesto** [comment]
FWIW, I wasn't sure about the cite attribute either so I looked for the spec: https://html.spec.whatwg.org/multipage/grouping-content.html#attr-blockquote-cite . It's optional and if it exists, it's meant to be a URL that cites the work. User agents may allow users to follow such citation links... but I don't think any browsers have implemented this

**Comment by canerakdas** [comment]
> FWIW, I wasn't sure about the cite attribute either so I looked for the spec: [html.spec.whatwg.org/multipage/grouping-content.html#attr-blockquote-cite](https://html.spec.whatwg.org/multipage/grouping-content.html#attr-blockquote-cite) . It's optional and if it exists, it's meant to be a URL that cites the work. User agents may allow users to follow such citation links... but I don't think any browsers have implemented this

Thank you, it is explained much better here. So should we keep this prop in the component?

**Comment by ovflowd** [comment]
The prop? No. Markdown has no cite syntax. https://stackoverflow.com/a/2002150

Hence I spoke about styling a children cite element, which you did on the PR

**Comment by ovflowd** [comment]
Now I'm only unsure if every Blockquote should have semibold text that is large by size. I think for now we can keep as it is, but we can always change if needed.

**Review by mikeesto — APPROVED** [review]
LGTM. The font weight can always be changed in the future

**Comment by ovflowd** [comment]
@canerakdas can you just remove the cite prop then? It has no usability for now. We can always add it if needed in the future.

**Pull request merged**